doc/sphinx/language/core/records.rst

```diff
@@ -80,36 +148,46 @@ in which case the correctness of :n:`@type__3` may rely on the instance :n:`@ter
    Note here that the fields ``Rat_bottom_cond`` depends on the field ``bottom``
    and ``Rat_irred_cond`` depends on both ``top`` and ``bottom``.
 
-Let us now see the work done by the :cmd:`Record` command. First the command
-generates a variant type definition with just one constructor:
-:n:`Variant @ident {* @binder } : @sort := @ident__0 {* @binder }`.
-
-To build an object of type :token:`ident`, provide the constructor
-:n:`@ident__0` with the appropriate number of terms filling the fields of the record.
-
 .. example::
 
-   Let us define the rational :math:`1/2`:
+   Let us define the rational :math:`1/2` using either the applicative or record syntax:
 
     .. coqtop:: in
 
        Theorem one_two_irred : forall x y z:nat, x * y = 1 /\ x * z = 2 -> x = 1.
        Admitted.
 
+       (* Applicative form: values for fields should be given in order *)
        Definition half := mkRat true 1 2 (O_S 1) one_two_irred.
-       Check half.
 
-Alternatively, the following syntax allows creating objects by using named fields, as
-shown in this grammar. The fields do not have to be in any particular order, nor do they have
-to be all present if the missing ones can be inferred or prompted for
-(see :ref:`programs`).
+       (* Record form: omitting inferrable fields top and bottom *)
+       Definition half' :=
+         {| sign := true;
+            Rat_bottom_cond := O_S 1;
+            Rat_irred_cond := one_two_irred |}.
 
-.. coqtop:: all
+.. example::
 
-  Definition half' :=
-    {| sign := true;
-       Rat_bottom_cond := O_S 1;
-       Rat_irred_cond := one_two_irred |}.
+   Let us define a function by pattern matching over a record:
+
+   .. coqtop:: all
+
+      Eval compute in (
+        match half with
+        | {| sign := true; top := n |} => n
+        | _ => 0
+        end).
+
+.. example::
+
+   Let us project fields of a record, using either the applicative or projection syntax:
+
+   .. coqtop:: all
+
+      Eval compute in top half.
+      Eval compute in bottom half.
+      Eval compute in Rat_bottom_cond half.
+      Eval compute in half.(top).
```

Comment:
Isn't there a simpler way to extract the value of a field that doesn't require `compute` or `Eval`?

This doesn't work--shouldn't it?

```
Goal True.
let x := half.top in idtac x.
```

Reply:
Also I don't see how `Eval compute in top half.` produces `1`.  `top half` is 1 but I don't get what `compute` does with that.  Quite an opaque example.